pilot/pkg/model/push_context.go

```diff
@@ -666,8 +666,22 @@ func (ps *PushContext) VirtualServices(proxy *Proxy, gateways map[string]bool) [
 			}
 		}
 	}
-
-	return out
+	if proxy == nil {
+		return out
+	}
+	// Make virtual services in the config namespace appear first
```

Comment:
may be restrict this to just sidecars and not gateways ?

Reply:
Any reason they should behave differently ?

Reply:
I mean: Gateways should also pick same-namespace ( istio-system or whatever ) first.